fv3gfs/util/communicator.py

```diff
@@ -451,6 +452,11 @@ def halo_update(self, quantity: Quantity, n_points: int):
         req = self.start_halo_update(quantity, n_points)
         req.wait()
 
+    @staticmethod
+    def _device_synchronize():
+        """Wait for all work that could be in-flight to finish."""
+        device_synchronize()
```

Comment:
This context would be good to have in a comment so this doesn't get refactored out by someone else.

```suggestion
        """Wait for all work that could be in-flight to finish."""
        # this is a method so we can profile it separately from other device syncs
        device_synchronize()
```